usb.go

```diff
@@ -0,0 +1,53 @@
+// hid - Gopher Interface Devices (USB HID)
+// Copyright (c) 2019 Péter Szilágyi, Guillaume Ballet. All rights reserved.
+//
+// This file is released under the 3-clause BSD license. Note however that Linux
+// support depends on libusb, released under GNU LGPL 2.1 or later.
+
+// Package usb provide interfaces for generic USB devices.
+package hid
+
+// DeviceType represents the type of a USB device (generic or HID)
+type DeviceType int
+
+// List of supported device types
+const (
+	DeviceTypeGeneric DeviceType = 0
+	DeviceTypeHID     DeviceType = 1
+)
+
+// Enumerate returns a list of all the HID devices attached to the system which
+// match the vendor and product id:
+//  - If the vendor id is set to 0 then any vendor matches.
+//  - If the product id is set to 0 then any product matches.
+//  - If the vendor and product id are both 0, all HID devices are returned.
+// func Enumerate(vendorID uint16, productID uint16) []DeviceInfo {
+// }
+
+// DeviceInfo is a generic libusb info interface
+type DeviceInfo interface {
+	// Type returns the type of the device (generic or HID)
+	Type() DeviceType
```

Comment:
I'm not sure making DeviceInfo and Device interfaces is a good change. That's not to say we shouldn't do it if there are solid reasons for it. Here's my thinking: the fields in HidDeviceInfo and GenericDeviceInfo are very similar and could just share the struct. Generic devices have less info and would just leave the associated fields at zero. To make Open work we could just add a private field specifying the backend.

Reply:
The reason is that every client code would then have to make the difference between a HID device and a generic device, based on what the content of the fields is. Then there's the question of the stuff that should be available in HID but won't be, like `Endpoints`. Guaranteeing that is a lot of work, and I'm not ready to do it because I see minimal benefit to this (geth won't use it)